### PR TITLE
Gnome eapi updates

### PR DESCRIPTION
**Pull request by SoapGentoo**

### CODE REVIEW

**Review by arthurzam — APPROVED** [review]
Looks good to me. Here is a list of extra changes to bring eclass to better QA. Feel free to ignore me.

`gnome.org`:

1. line 39, use `${PN}`

`gnome2-utils`:

1. line 102, may as well use simpler to read `if [[ ${EAPI} = 7 ]]; then`
2. Can you please change `GNOME2_ECLASS_SCHEMAS` to use array notation across the file (internal variable, so no need for backward compatibility)
3. Can you also modify `omf_makefiles` to use array?

`gnome2`:

1. Can you add in `GNOME2_EAUTORECONF` doc that the value to enable it is `yes`?
2. the line `ELTCONF=${ELTCONF:-""}` feels weird to me. Isn't it equal to just omitting it?
3. I might be wrong here, but I think the read loop in `gnome2_pkg_preinst` can be simplified with readarray in the newer bash supported by EAPIs.

File: eclass/gnome2.eclass [hunk 1]
```diff
@@ -170,16 +162,8 @@ gnome2_src_install() {
...
-			yes)    prune_libtool_files --modules;;
-			no)     ;;
-			*)      prune_libtool_files;;
-		esac
-	else
-		if [[ ${GNOME2_LA_PUNT} != 'no' ]]; then
-			find "${ED}" -name '*.la' -delete || die
-		fi
+	if [[ ${GNOME2_LA_PUNT} != no ]]; then
+		find "${ED}" -name '*.la' -delete || die
```
Comment (ulm) [inline eclass/gnome2.eclass:166]: Add `-type f` so it agrees with [PG 0303](https://projects.gentoo.org/qa/policy-guide/installed-files.html#pg0303).

```suggestion
		find "${ED}" -type f -name '*.la' -delete || die
```

**Review by arthurzam — APPROVED** [review]

File: eclass/gnome2-utils.eclass [hunk 2]
```diff
@@ -221,39 +220,37 @@ gnome2_gconf_uninstall() {
...
 	if [[ -f ${S}/gnome-doc-utils.make ]] ; then
-		omf_makefiles="${omf_makefiles} ${S}/gnome-doc-utils.make"
+		omf_makefiles+=( "${S}"/gnome-doc-utils.make )
 	fi
 
 	# testing fixing of all makefiles found
 	# The sort is important to ensure .am is listed before the respective .in for
 	# maintainer mode regeneration not kicking in due to .am being newer than .in
+	local filename
 	for filename in $(find "${S}" -name "Makefile.in" -o -name "Makefile.am" |sort) ; do
```
Comment (mgorny) [inline eclass/gnome2-utils.eclass:237]: while-read, since you're already making it whitespace friendly?

File: eclass/gnome2-utils.eclass [hunk 3]
```diff
@@ -221,39 +220,37 @@ gnome2_gconf_uninstall() {
...
+	for omf in "${omf_makefiles[@]}" ; do
 		sed -i -e 's:scrollkeeper-update:true:' "${omf}"
 		retval=$?
 
 		if [[ $retval -ne 0 ]] ; then
 			debug-print "updating of ${omf} failed"
 
 			# Add to the list of failures
-			fails[$(( ${#fails[@]} + 1 ))]=$omf
+			fails[$(( ${#fails[@]} + 1 ))]=${omf}
```
Comment (mgorny) [inline eclass/gnome2-utils.eclass:253]: Isn't that `fails+=( ... )`?

**Pull request closed without merging**